core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/OperatorUtils.java

```diff
@@ -104,6 +104,57 @@ public String toString() {
         };
   }
 
+  /**
+   * Construct {@link FunctionBuilder} which call function with arguments produced by observer1 and
+   * observer2 In general, if any operand evaluates to a MISSING value, the enclosing operator will
+   * return MISSING; if none of operands evaluates to a MISSING value but there is an operand
+   * evaluates to a NULL value, the enclosing operator will return NULL.
+   *
+   * @param functionName function name
+   * @param function     {@link BiFunction}
+   * @param observer1    extract the value of type T from the first argument
+   * @param observer2    extract the value of type U from the second argument
+   * @param returnType   return type
+   * @param <T>          the type of the first argument to the function
+   * @param <U>          the type of the second argument to the function
+   * @param <R>          the type of the result of the function
+   * @return {@link FunctionBuilder}
+   */
+  public static <T, U, R> FunctionBuilder doubleArgFunc(
```

Comment:
What's the difference between this new method and `binaryOperator()` above?

Reply:
only the `toString` method is different

Reply:
Okay. For now `toString` seems being used for column name in response, I think we can treat arithmetic and function toString same way later when column name is retrieved in other way.